### PR TITLE
LESQ-955: OakPagination component 

### PR DESCRIPTION
**Pull request by k-huggs**
# How to review this PR

_Leave this text block for the reviewer_

- Check [component hierarchy](https://miro.com/app/board/uXjVNnKBgyk=/?share_link_id=59445593794) is followed correctly
- Check the design [Heuristics](https://components.thenational.academy/?path=/docs/docs-howtodesigncomponents--docs#heuristics-for-component-design) have been followed
- Check [naming conventions](https://components.thenational.academy/?path=/docs/docs-namingconventions--docs) have been applied
- Check for these gotchyas:
  - Missing exports for Oak components
  - Accidental export of Internal components
  - Snapshots of unexpected components have been modified
  - Circular dependencies
  - Code duplication (via not using base components)
  - Non-functional storybook for this or related components
  - Sensitve files changed eg. atoms, or style tokens
  - Relative imports
  - Default exports

# Add your PR description below

#LESQ-955

Oak Pagination component to be utilised across OWA

## Link to the design doc

https://www.figma.com/design/YcWQMMhHPVVmc47cHHEEAl/Oak-Design-Kit?node-id=9613-1123&m=dev

## A link to the component in the deployment preview

https://deploy-preview-245--lively-meringue-8ebd43.netlify.app/?path=/docs/components-organisms-oakpagination--docs

### Functionality
- Chevrons can be used to increment and decrement pages
- Each number is a clickable link which can take you to the correct URL which is passed in 

### Style 
- If a page is selected it should indicate it's active (black bolding)
- As you scroll, when there are 8 or more pages, ellipse should shorten list to only include last page
- If the active page is 4 there should be a double ellipses on the start and end of the list
- If the active page is 5, single ellipses at the start of the menu after 1

## ACs

- [ ]  Page number links should be called “DOMAIN page x” (where domain is blogs/subject name etc)” for a11y
- [ ] “No previous/further pages” when chevrons are disabled
- [ ] Ellipses are invisible to keyboard users/screen readers
- [ ] Same on tablet/mobile

### CODE REVIEW

File: src/components/organisms/OakPagination/OakPagination.tsx [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+      >
+        <StyledChevronButton
+          element={isFirstPage ? "button" : "a"}
+          data-testid="backwards-button"
+          onClick={() => {
+            handleChevronClick("backwards");
+          }}
+          href={prevHref}
+          aria-disabled={isFirstPage}
+          disabled={activePage <= 1}
```
Comment (njblore) [inline src/components/organisms/OakPagination/OakPagination.tsx:148]: isFirstPage?

File: src/components/organisms/OakPagination/OakPagination.tsx [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+                  $mh={[
+                    "space-between-sssx",
+                    "space-between-ssx",
+                    "space-between-ssx",
+                  ]}
+                >
+                  <OakPageNumber
+                    pageName={pageName}
+                    key={pageIndex}
+                    pageIndex={pageIndex + 1}
```
Comment (njblore) [inline src/components/organisms/OakPagination/OakPagination.tsx:168]: it's a bit inconsistent how pageindex and pageNumber are named throughout, this should be pageNumber if it's the index +1? similarly we're mapping over indices but they're called numbers

Comment (k-huggs) [reply src/components/organisms/OakPagination/OakPagination.tsx]: ✅

**Review by njblore — APPROVED** [review]
nice 🎉

**Review by jack-skerrett-oak — DISMISSED** [review]
- [x] I mentioned in slack, but doing here for completeness that focus styles aren't appearing when tabbing through the component, and the chevrons don't seem to be clickable
- [x] The chevrons are links, but the numbers are buttons. I think it would be best if they were all links because links are for nav, buttons are for actions (https://a11y-101.com/design/button-vs-link#:~:text=A%20link%20will%20redirect%20you,like%20opening%20or%20closing%20something.)
- [x] Could "rel='next|prev'" be added to the previous and next chevrons to make it clear what their relationship is to the current page

**Comment by RBrownsett** [comment]
@k-huggs possible issue when viewing the pagination in small mobile mode in storybook - the left chevron is pushed off the screen:
image

**Review by Monika-Koziol — CHANGES_REQUESTED** [review]
- [x] when clicking through the selected page has an additional blue border, wich it shouldn't have
- [ ] for all state changes for the buttons please follow the styles of link in figma

**Review by RBrownsett — APPROVED** [review]
Approved - mobile view is now working as expected.

Re-tested tabbing and focus is shown correctly. Buttons now removed and links used. Screenreader tested successfully.

**Comment by k-huggs** [comment]
> @k-huggs possible issue when viewing the pagination in small mobile mode in storybook - the left chevron is pushed off the screen: image



> @k-huggs possible issue when viewing the pagination in small mobile mode in storybook - the left chevron is pushed off the screen: image

https://github.com/user-attachments/assets/9bcfa845-4d09-48cc-91dc-9fbd38b55f45

**Review by Monika-Koziol — CHANGES_REQUESTED** [review]
@k-huggs Sorry, I didn't specify which link in my last comment. Can the numbers use our standard black rather than navy? Other than that all good

**Review by Monika-Koziol — APPROVED** [review]

**Pull request merged**